storage/postgres/tree_storage.go

```diff
@@ -41,7 +41,7 @@ const (
 			FROM subtree n
 			WHERE n.subtree_id IN (` + placeholderSQL + `) AND
 			n.tree_id = ? AND n.subtree_revision <= ?
-			GROUP BY n.subtree_id
+			GROUP BY n.tree_id, n.subtree_id
```

Comment:
After doing a bit of research into Postgres, I've found that [it doesn't support loose index scans](https://wiki.postgresql.org/wiki/Loose_indexscan), which is what this optimisation takes advantage of, so making this change to the Postgres SQL doesn't achieve anything. It appears that it may be possible to modify this SQL to use a recursive CTE (Common Table Expression), which would achieve the same optimisation, but I'm not sure it's worth going to the trouble right now. Perhaps I should just leave a TODO here to consider it?

Reply:
Yeah, TODO seems like the best idea for now.

Reply:
Done.